dev-tools/oxlog/src/lib.rs

```diff
@@ -0,0 +1,442 @@
+        // for all zones on external storage pools.
+        let pools = Pools::read()?;
+        for uuid in &pools.external {
+            let zones_path: Utf8PathBuf =
+                ["/pool/ext", &uuid.to_string(), "crypt/zone"].iter().collect();
+            // Find the zones on the given pool
+            let Ok(entries) = read_dir(zones_path.as_path()) else {
+                continue;
+            };
+            for entry in entries {
+                let Ok(zone_entry) = entry else {
+                    continue;
+                };
+                let zone = zone_entry.file_name();
+                let Some(zone) = zone.to_str() else {
+                    // not utf8
+                    continue;
+                };
+                // Add the path to the current logs for the zone
+                let mut dir = zones_path.clone();
+                dir.push(zone);
+                dir.push("root/var/svc/log");
+                let mut paths =
+                    Paths { primary: dir, debug: vec![], extra: vec![] };
+
+                // Add the path to the extra logs for the zone
+                if zone.starts_with("oxz_cockroachdb") {
+                    let mut dir = zones_path.clone();
+                    dir.push(zone);
+                    dir.push("root/data/logs");
+                    paths.extra.push(("cockroachdb", dir));
+                }
+
+                zones.insert(zone.to_string(), paths);
+            }
+        }
+
+        // Find the directories containing the debug log files
+        for uuid in &pools.external {
+            let zones_path: Utf8PathBuf =
+                ["/pool/ext", &uuid.to_string(), "crypt/debug"]
+                    .iter()
+                    .collect();
+            // Find the zones on the given pool
+            let Ok(entries) = read_dir(zones_path.as_path()) else {
+                continue;
+            };
+            for entry in entries {
+                let Ok(zone_entry) = entry else {
+                    continue;
+                };
+                let zone = zone_entry.file_name();
+                let Some(zone) = zone.to_str() else {
+                    // not utf8
+                    continue;
+                };
+                let mut dir = zones_path.clone();
+                dir.push(zone);
+
+                // We only add debug paths if the zones have primary paths
+                if let Some(paths) = zones.get_mut(zone) {
+                    paths.debug.push(dir);
+                }
+            }
+        }
+
+        Ok(Zones { zones })
+    }
+
+    /// Return log files organized by service name
+    pub fn zone_logs(
+        &self,
+        zone: &str,
+        filter: Filter,
+    ) -> BTreeMap<ServiceName, SvcLogs> {
+        let mut output = BTreeMap::new();
+        let Some(paths) = self.zones.get(zone) else {
+            return BTreeMap::new();
+        };
+        load_svc_logs(paths.primary.clone(), &mut output);
```

Comment:
I specifically did not add a filter check before this line because:
 1. This is fast
 2. Logs that have been rotated but not yet moved into the `debug` directory will be added to archive paths, so I didn't want to only filter on `current`. 

However, I now realize I could filter such that we don't load the logs if `filter.current` and `filter.archived` are both false.

Reply:
Fixed in [f2b2f8a](https://github.com/oxidecomputer/omicron/pull/4810/commits/f2b2f8a245aebd32243937a72ae2ef0603f15f15)